File: app/src/main/java/com/absoluteapps/arthurl/mobooru/Formatter.java
package com.absoluteapps.arthurl.mobooru;

public class Formatter {

    // Converts and rounds:
    // 1234 = 1.2k
    // 9876543 = 9.9M
    public static String shortHandFormatter(int value) {
        if (value < 1000) {
            return value + " ";
        } else if (value < 1000000) {
            return Math.round(value / 100) / (1.0 * 10) + "k ";
        } else {
            return Math.round(value / 100000) / (1.0 * 10) + "M ";
        }
    }
}

File: app/src/main/java/com/absoluteapps/arthurl/mobooru/Formatter.kt
package com.absoluteapps.arthurl.mobooru

import kotlin.math.roundToInt

object Formatter {

    // Converts and rounds:
    // 1234 = 1.2k
    // 9876543 = 9.9M
    fun shortHandFormatter(value: Int): String {
        return if (value < 1000) {
            "$value "
        } else if (value < 1000000) {
            ((value / 100).toFloat().roundToInt() / (1.0 * 10)).toString() + "k "
        } else {
            ((value / 100000).toFloat().roundToInt() / (1.0 * 10)).toString() + "M "
        }
    }
}
